### PR TITLE
Marcos.cifuentes/feature pokemon card details activity ui

### PR DESCRIPTION
**Pull request by MarcosCGlobant**
In this PR I implemented the UI for the card detail activity feature, the information shown is mocked, because the next PR will be the service implementation.

- Modified card list activity to add onClickListeners to the elements.
- Added Card Detail implementation with mocked data.
- Refactored Pokemon Card Response and Pokemon Card entity to recover the detail data
- New and modified unit tests for the view models.

GIF of the app:

card detail ui

Unit test passing and coverage:

card detail test passed

details ui coverage

### CODE REVIEW

File: app/src/main/java/com/globant/pokemontcgapp/viewmodel/PokemonCardDetailViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+class PokemonCardDetailViewModel :
+    ViewModel(), PokemonCardDetailContract.ViewModel {
+
+    private val pokemonCardMutableLiveData = MutableLiveData<Data>()
+    override fun getPokemonCardLiveData(): LiveData<Data> = pokemonCardMutableLiveData
+
+    override fun getPokemonCard() = viewModelScope.launch {
+        // TODO: Recover the card information from the API using a use case
+        pokemonCardMutableLiveData.value = Data(Status.SUCCESS, MockedPokemonCard().pokemonCardSelected)
```
Comment (lucianoeli) [inline app/src/main/java/com/globant/pokemontcgapp/viewmodel/PokemonCardDetailViewModel.kt]: elminate this MockedPokemonCard(), instantiate the entity

File: app/src/main/java/com/globant/pokemontcgapp/activity/PokemonCardDetailActivity.kt [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+    override fun onResume() {
+        super.onResume()
+        binding.activityPokemonCardDetailLoader.visibility = View.GONE
+        pokemonCardDetailViewModel.getPokemonCard()
+    }
+
+    private fun updateUI(pokemonCardDetailData: Data) {
+        when (pokemonCardDetailData.status) {
+            Status.SUCCESS -> showPokemonCardDetails(pokemonCardDetailData.data)
```
Comment (lucianoeli) [inline app/src/main/java/com/globant/pokemontcgapp/activity/PokemonCardDetailActivity.kt:42]: you should add the loading state

File: app/src/main/java/com/globant/pokemontcgapp/viewmodel/PokemonCardDetailViewModel.kt (hunk 1 above)
Comment (MarcosCGlobant) [reply app/src/main/java/com/globant/pokemontcgapp/viewmodel/PokemonCardDetailViewModel.kt]: done, thanks

File: app/src/main/java/com/globant/pokemontcgapp/activity/PokemonCardDetailActivity.kt (hunk 2 above)
Comment (MarcosCGlobant) [reply app/src/main/java/com/globant/pokemontcgapp/activity/PokemonCardDetailActivity.kt:42]: added, thanks

**Review by SartiMau — APPROVED** [review]

**Review by SartiMau — APPROVED** [review]

**Review by lucianoeli — APPROVED** [review]

**Review by hernanhorquin96 — APPROVED** [review]

**Review by mauroLonghi — APPROVED** [review]

**Review by nicolbidone — APPROVED** [review]

**Pull request merged**